app_server/test/controllers/SubscriptionController.js

```diff
@@ -0,0 +1,400 @@
+      function(user1, user2, user3) {
+        testAccount.userId = user1.userId;
+        Router.inject({method: 'POST',
+                        url: '/api/subscriptions/' + user2.userId,
+                        credentials: testAccount}, (res) => {
+          subscribePromise(user3);
+        });
+      });
+  });
+
+  lab.test('Get subscriptions valid empty', function(done) {
+    var userPromise1 = Service.createNewUser(bob);
+
+    userPromise1.then(function(user) {
+      testAccount.userId = user.userId;
+      Router.inject({method: 'GET',
+                     url: '/api/subscriptions',
+                     credentials: testAccount}, function(res) {
+        Code.expect(res.result).deep.equal([]);
+        done();
+      });
+    });
+  });
+
+  lab.test('Get subscriptions invalid userId', function(done) {
+    testAccount.userId = '3388ffff-aa00-1111a222-00000044888c';
+    Router.inject({method: 'GET',
+                   url: '/api/subscriptions',
+                   credentials: testAccount}, function(res) {
+      Code.expect(res.result.statusCode).to.equal(400);
+      Code.expect(res.result.message).to.equal('User not found');
+      done();
+    });
+  });
+
+  lab.test('Get subscribers valid', function(done) {
+    var userPromise1 = Service.createNewUser(bob);
+    var userPromise2 = Service.createNewUser(alice);
+    var userPromise3 = Service.createNewUser(carlos);
+
+    // Second subscription
+    function subscribePromise(user2, user3) {
+      testAccount.userId = user3.userId;
+      Router.inject({method: 'POST',
+                     url: '/api/subscriptions/' + user2.userId,
+                     credentials: testAccount}, (res) => {
+        querySubscribers(user2);
+      });
+    }
+
+    function querySubscribers(user2) {
+      Router.inject({method: 'GET',
+                     url: '/api/subscriptions/' + user2.userId,
+                     credentials: testAccount}, function(res) {
+        Code.expect(res.result).to.have.length(2);
+        Code.expect(res.result[0].username).to.equal(bob.username);
+        Code.expect(res.result[1].username).to.equal(carlos.username);
+        done();
+      });
+    }
+
+    // First subscription
+    Promise.join(userPromise1, userPromise2, userPromise3,
+      function(user1, user2, user3) {
+        testAccount.userId = user1.userId;
+        Router.inject({method: 'POST',
+                        url: '/api/subscriptions/' + user2.userId,
+                        credentials: testAccount}, (res) => {
+          subscribePromise(user2, user3);
+        });
+      });
+
+    /*    Promise.join(userPromise1, userPromise2, userPromise3,
+      function(user1, user2, user3) {
+        testAccount.userId = user1.userId;
+         Router.inject({method: 'POST',
+                        url: '/api/subscriptions/' + user2.userId,
+                        credentials: testAccount},
+                        subscribePromise(null, user2, user3));
+      });*/
+  });
```

Comment:
Remove unused codes instead of commenting out. Version control can be used to archive and revise old/discarded code :-P